### PR TITLE
Add the exhaustiveNbHits in search response body (returns always false)

### LINKED ISSUE #103: Missing exhaustiveNbHits in search response
To be iso, the response body of the search (GET and POST) should contain the `exhaustiveNbHits` boolean.

See docs of the search response: https://docs.meilisearch.com/reference/api/search.html#response

### PR DESCRIPTION
**Pull request by irevoire**
closes #103

### CODE REVIEW

**Comment by curquiza** [comment]
@MarinPostma does MeiliSearch (the old one) returns sometimes another value than `false`?

**Comment by MarinPostma** [comment]
No never

File: meilisearch-http/src/index/search.rs [hunk 1]
```diff
@@ -41,6 +41,7 @@ pub struct SearchQuery {
 pub struct SearchResult {
     pub hits: Vec<Map<String, Value>>,
     pub nb_hits: u64,
+    pub exhaustive_nb_hits: bool, // currently this field only exist to be ISO and is always alse
```
Comment (MarinPostma) [inline meilisearch-http/src/index/search.rs:44]:
```suggestion
    pub exhaustive_nb_hits: bool, // currently this field only exist to be ISO and is always false
```

**Review by MarinPostma — CHANGES_REQUESTED** [review]

File: meilisearch-http/src/index/search.rs [hunk 2]
```diff
@@ -107,6 +108,7 @@ impl Index {
         };
 
         let result = SearchResult {
+            exhaustive_nb_hits: false, // not implemented, we use it to be ISO
```
Comment (curquiza) [inline meilisearch-http/src/index/search.rs:111]:
```suggestion
            exhaustive_nb_hits: false, // not implemented yet
```

Comment (curquiza) [reply meilisearch-http/src/index/search.rs:111]: We should not talk about being ISO in the code since this code will be merged into MeiliSearch 🙂

**Review by curquiza — CHANGES_REQUESTED** [review]

**Review by curquiza — APPROVED** [review]

**Review by curquiza — CHANGES_REQUESTED** [review]
Sorry to ask you this now -> can you add a check in one of the tests to check the field `exhaustiveNbHits` is present in the search response body?

**Comment by curquiza** [comment]
What do you mean?

I agree we do not need to test the value of exhausitveNbHits 🙂 but in any cases the exhausitveNbHits key should be present in the search response. So testing the key is present could be great

**Comment by MarinPostma** [comment]
@curquiza, what @irevoire  means is that we haven't started testing search just yet :)

We are waiting for milli to stabilize the search before we get to it.

@irevoire, can you add the tests we have to do to #3 so we don't forget them? :)

File: meilisearch-http/src/index/search.rs [hunk 3]
```diff
@@ -41,6 +41,7 @@ pub struct SearchQuery {
 pub struct SearchResult {
     pub hits: Vec<Map<String, Value>>,
     pub nb_hits: u64,
+    pub exhaustive_nb_hits: bool, // currently this field only exist to be ISO and is always false
```
Comment (MarinPostma) [inline meilisearch-http/src/index/search.rs:44]:
```suggestion
    pub exhaustive_nb_hits: bool,
```

**Review by MarinPostma — CHANGES_REQUESTED** [review]
One last change, given @curquiza 's last comment

**Review by MarinPostma — APPROVED** [review]
bors merge

**Comment by curquiza** [comment]
Got it!

**Review by curquiza — APPROVED** [review]

**Comment by MarinPostma** [comment]
bors merge

**Comment by curquiza** [comment]
We could not merge with bors because of me (in the settings). I fixed it! Sorry

bors merge

**Pull request merged**